### PR TITLE
Add secure_env_removed to payload

### PR DESCRIPTION
**Pull request by BanzaiMan**
travis-build will look at this information, and warn when PR comes
from a fork and secure environment variables are not available.

### CODE REVIEW

**Comment by BanzaiMan** [comment]
This PR supersedes #18.

**Comment by BanzaiMan** [comment]
Both Repository settings and `.travis.yml` contents are considered when determining whether secure variables are removed from the build request.

File: lib/travis/scheduler/record/repository/settings.rb [hunk 1]
```diff
@@ -122,6 +122,10 @@ def timeout_log_silence
   def repository_id
     additional_attributes[:repository_id]
   end
+
+  def has_secure_var?
```
Comment (svenfuchs) [inline lib/travis/scheduler/record/repository/settings.rb]: Should this be plural? `has_secure_vars?` The serializer also uses plural: `secure_env_vars_removed?`

File: lib/travis/scheduler/serialize/worker/job.rb [hunk 2]
```diff
@@ -30,6 +30,15 @@ def same_repo_pull_request?
             request.same_repo_pull_request?
           end
 
+          def secure_env_vars_removed?
+            !secure_env? &&
+            job.repository.settings.has_secure_var? || [:env, :global_env].any? do |key|
+              config.has_key?(key) &&
```
Comment (svenfuchs) [inline lib/travis/scheduler/serialize/worker/job.rb]: I usually use just `key?` on Hashes.

File: spec/travis/scheduler/serialize/worker_spec.rb [hunk 3]
```diff
@@ -122,6 +123,7 @@ def encrypted(value)
             pull_request: false,
             state: 'queued',
             secure_env_enabled: true,
+            secure_env_vars_removed: false,
```
Comment (svenfuchs) [inline spec/travis/scheduler/serialize/worker_spec.rb]: Looking at this I am wondering if we should pass just `secure_env_removed`, as it's also `secure_env_enabled`?

**Comment by svenfuchs** [comment]
@BanzaiMan Looks good 👍  LGTM

**Pull request merged**